libraries/stdlib/osl/stdosl.h

```diff
@@ -601,13 +608,32 @@ closure color burley_diffuse_bsdf(normal N, color albedo, float roughness) BUILT
 // a VDF closure describing the surface interior to handle absorption and scattering
 // inside the medium.
 //
+//  \param  N                 Normal vector of the surface point beeing shaded.
+//  \param  U                 Tangent vector of the surface point beeing shaded.
+//  \param  reflection_tint   Weight per color channel for the reflection lobe. Should be (1,1,1) for a physically-correct dielectric surface, 
+//                            but can be tweaked for artistic control. Set to (0,0,0) to disable reflection.
+//  \param  transmission_tint Weight per color channel for the transmission lobe. Should be (1,1,1) for a physically-correct dielectric surface, 
+//                            but can be tweaked for artistic control. Set to (0,0,0) to disable transmission.
+//  \param  roughness_x       Surface roughness in the U direction. Valid range [0,1] with a perceptually linear response over the range.
+//  \param  roughness_y       Surface roughness in the V direction. Valid range [0,1] with a perceptually linear response over the range.
+//  \param  ior               Refraction index.
+//  \param  distribution      Microfacet distribution. An implementation is expected to support the following distributions: { "ggx" }
+//
 closure color dielectric_bsdf(normal N, vector U, color reflection_tint, color transmission_tint, float roughness_x, float roughness_y, float ior, string distribution) BUILTIN;
 ​
 // Constructs a reflection BSDF based on a microfacet reflectance model.
 // Uses a Fresnel curve with complex refraction index for conductors/metals.
 // If an artistic parametrization is preferred the artistic_ior() utility function
 // can be used to convert from artistic to physical parameters.
 //
+//  \param  N                 Normal vector of the surface point beeing shaded.
+//  \param  U                 Tangent vector of the surface point beeing shaded.
+//  \param  roughness_x       Surface roughness in the U direction. Valid range [0,1] with a perceptually linear response over the range.
```

Comment:
Some anisotropic parameterizations give roughness values beyond 1 given a roughness/anisotropy factor in `[0,1]`. For example:

```
alpha_x = roughness^2 * (1 - anisotropy)
alpha_y = roughness^2 * (1 + anisotropy)
```

The alpha values range from `[0,2]` so the corresponding `roughness_x/y` range from `[0,sqrt(2)]` right ?

The anisotropic parameterization in the disney model also goes >1 for the individual roughnesses.

So maybe the bit about valid range could just be left out ?

Reply:
For the example above I think roughness_x/y should still be [0,1], right? If it's [0,sqrt(2)] you would get the alpha values range [0,4], unless I'm missing something.

But in general it feels like we should be able to have a defined range [0,1] where 0 means perfectly smooth and 1 means fully diffuse, and the implementation should try to follow this behavior, in the best way possible depending on the distribution used. And independent from an anisotropic parameterization, if used on the outside.

Reply:
`roughness` and `anisotropy` are both in `[0,1]`. Therefore `alpha_x` is in `[0,1]` but `alpha_y` is in `[0,2]`.

Assuming that `roughness_y=sqrt(alpha_y)`, then the range for that one is `[0,sqrt(2)]`.

I'm not sure this really impacts much since the "requirement" of the values being in `[0,1]` is more of a suggestion, but just mentioning it in case.

I am curious how many other parameterizations of anisotropy are out there. I believe there is mostly this one and the disney model:

```
    float aspect = sqrt(1-anisotropic*.9);
    float ax = sqr(roughness)/aspect;
    float ay = sqr(roughness)*aspect;
```

The disney approach (if I am doing my math right) has an upper bound for `roughness_x` of ~1.778.

Reply:
Right, as long as the user-facing roughness is in [0,1] the ranges used under the hood is less important. I will update the doc strings.

For anisotropic parameterizations, the one you describe above is the only one we use in the MaterialX library. It's the same one being used in Autodesk Standard Surface.